File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/OperationUpdates.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.*;

@SuppressWarnings("unused")
public class OperationUpdates {
  Fragment fragment = Fragment.of("DELETE FROM users WHERE last_login < now() - interval '1 year'");

  // start
  // Get the affected row count
  Operation<Integer> rowCount = fragment.update();

  // Ignore the row count (DDL, fire-and-forget DML)
  Operation<Void> ignored = fragment.execute();
  // stop
}

File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/core/OperationUpdates.kt
package dev.typr.foundationskt.docs.core

import dev.typr.foundationskt.*

@Suppress("unused")
class OperationUpdates {
    val fragment: Fragment = sql {
        "DELETE FROM users WHERE last_login < now() - interval '1 year'" }

    //start
    // Get the affected row count
    val rowCount: Operation<Int> = fragment.update()

    // Ignore the row count (DDL, fire-and-forget DML)
    val ignored: Operation<Unit> = fragment.execute()
    //stop
}
